### PR TITLE
[FAU-445] Restrict search to pre-selected degree programs

### PR DESCRIPTION
**Pull request by zhyian**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)
https://inpsyde.atlassian.net/browse/FAU-445

**What is the new behavior (if this is a feature change)?**
Pre-selected filters are now added as hidden inputs, ensuring that all they are included in form submissions without any additional  handling.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:
Additionally, I updated several internal packages: `inpsyde/assets`, `inpsyde/modularity`, `inpsyde/wp-context`. These updates fix text domain loading issues and resolve other problems.

### CODE REVIEW

File: src/Infrastructure/Component/PreAppliedFilters.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public function render(array $attributes = self::DEFAULT_ATTRIBUTES): string
+    {
+        /** @var PreAppliedFilters $attributes */
+        $attributes = wp_parse_args($attributes, self::DEFAULT_ATTRIBUTES);
+        $preAppliedFilters = [];
+
+        foreach ($attributes['preAppliedFilters'] as $name => $values) {
+            foreach ($values as $value) {
```
Comment (o-samaras) [inline src/Infrastructure/Component/PreAppliedFilters.php:32]: Sorry if this is obvious, but how do we ensure that `$values` is iteratable? Can't it be a simple string or numeric value?

Comment (zhyian) [reply src/Infrastructure/Component/PreAppliedFilters.php:32]: Thanks for pointing this out, the `$values` variable is always an array:

https://github.com/RRZE-Webteam/FAU-Studium-Embed/blob/74dac24a6c7a0c2e26cbfaa6daccba94f5561448/src/Infrastructure/Shortcode/ShortcodeAttributesNormalizer.php#L92-L100

https://github.com/RRZE-Webteam/FAU-Studium-Embed/blob/74dac24a6c7a0c2e26cbfaa6daccba94f5561448/src/Infrastructure/Shortcode/ShortcodeAttributesNormalizer.php#L115

**Review by o-samaras — APPROVED** [review]
Thank you for working on this. Code-wise LGTM

**Review by amiut — APPROVED** [review]
Thanks for taking care of it, LGTM

**Review by shvlv — APPROVED** [review]
Nice!

**Pull request merged**